### PR TITLE
Hs rework failure handling

### PR DESCRIPTION
**Pull request by hendrik-skubch**

### CODE REVIEW

File: alica_engine/src/engine/Assignment.cpp [hunk 1]
```diff
@@ -117,14 +117,16 @@ bool Assignment::isSuccessful() const
     }
     const EntryPointGrp& eps = _plan->getEntryPoints();
     const int numEps = eps.size();
+    bool ret = false;
     for (int i = 0; i < numEps; ++i) {
         if (eps[i]->isSuccessRequired()) {
-            if (_successData.getRaw()[i].empty() || static_cast<int>(_successData.getRaw()[i].size()) < eps[i]->getMinCardinality()) {
+            if (!(_successData.getRaw()[i].empty() && static_cast<int>(_successData.getRaw()[i].size()) >= eps[i]->getMinCardinality())) {
```
Comment (abolinsky) [inline alica_engine/src/engine/Assignment.cpp]: This is now incorrect. If _successData.getRaw()[i] is not empty, then isSuccessful will always return false.

File: alica_engine/src/engine/RuleBook.cpp [hunk 2]
```diff
@@ -103,6 +103,9 @@ PlanChange RuleBook::visit(RunningPlan& r)
             doDynAlloc = false;
         }
         changeRecord = updateChange(changeRecord, planAbortRule(r));
+        if (changeRecord == PlanChange::FailChange) {
```
Comment (abolinsky) [inline alica_engine/src/engine/RuleBook.cpp]: does a similar check need to be done for the other updateChange calls?

File: alica_engine/src/engine/Assignment.cpp (hunk 1 above)
Comment (hendrik-skubch) [reply alica_engine/src/engine/Assignment.cpp]: good find. Seems to be a bad merge

File: alica_engine/src/engine/RuleBook.cpp (hunk 2 above)
Comment (hendrik-skubch) [reply alica_engine/src/engine/RuleBook.cpp]: No, not at the moment. Though I don't particular like the current setup.
A more thorough refactoring might be in order in the future.

File: alica_engine/src/engine/RunningPlan.cpp [hunk 3]
```diff
@@ -104,6 +105,17 @@ bool RunningPlan::isDeleteable() const
         return true; // shortcut for plans from planselector
     return isRetired() && (!isBehaviour() || !_ae->getBehaviourPool()->isBehaviourRunningInContext(*this));
 }
+
+void RunningPlan::preTick()
+{
+    if (isRetired()) {
+        return;
+    }
+    evalRuntimeCondition();
```
Comment (gauthamm) [inline alica_engine/src/engine/RunningPlan.cpp]: This can throw std::exception, shouldn't we need to Abort or have a proper handling for exception?

Comment (hendrik-skubch) [reply alica_engine/src/engine/RunningPlan.cpp]: Hm? An exception during evaluation is caught in Condition::evaluate and should not reach this point.

**Pull request merged**